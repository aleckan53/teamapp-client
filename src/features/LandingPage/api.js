import config from 'config'
import TokenService from 'services/TokenService'

export default {
  postLogin(credentials, setLoading) {
    setLoading(true)
    return fetch(`${config.API_ENDPOINT}/auth/login`, {
      method: 'POST',
      body: JSON.stringify(credentials),
      headers: {
        'content-type': 'application/json',
      }})
      .then(res => {
        setLoading(false)
        if(!res.ok) {
          return res.json().then(e => Promise.reject(e))
        }

        return res.json()
      })
  },
  guestLogin(setAuthorized, history, setLoading) {
    this.postLogin({
      email: config.GUEST_EMAIL,
      password: config.GUEST_PASS,
    }, setLoading)
      .then(res => {
        TokenService.saveAuthToken(res.authToken)
        setAuthorized(true)
        history.push('/account')
      })
  }
}