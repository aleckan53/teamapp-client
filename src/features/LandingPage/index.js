import React, { useContext, useState } from 'react'
import styles from './styles.module.css'
import Api from './api'
import context from 'context'
import { BtnLink, Btn, LoaderFull } from '../Basic'

const LandingPage = props => {

  const [loading, setLoading] = useState(false)
  const { setAuthorized } = useContext(context.Events)

  const handleGuestLogin = () => {
    Api.guestLogin(setAuthorized, props.history, setLoading)
  }

  return  <>
    {loading ? <LoaderFull/> : ''}
    <div className={styles.landingPage}>
      <header className={styleMedia.header}>
        <div className={styles.hero}>
        </div>
        <span className={styles.version}>v 0.1 <strong>Beta</strong></span>
        <div className={styles.btnContainer}>
          <BtnLink 
            style={{color: 'var(--sub-text)'}}
            to='signup'
            className={styles.btn}
            type='button'
            title='Sign up'/>
          <BtnLink
            to='/login' 
            className={styles.btn}
            type='button'
            title='Log in'/>
        </div>
        <div className={styles.titleContainer}>
          <h1>A new way to collaborate!</h1>
          <h2>Welcome to the TeamApp</h2>
        </div>
      </header>
      <div>
        <section>
          <h3>About</h3>
          <p>
            TeamApp is a new collaboration platform, helping people to team up and work together.<br/><br/>
          </p>
          <img src={require('./img/ipad.png')} alt='ipad demo'/>
          <p>
            If you want to:
          </p>
            <ul>
              <li>Bring your great idea into life</li>
              <li>Volunteer to an open source project</li>
              <li>Gain a real working experience</li>
              <li>Level up your skills</li>
            </ul>
          <p style={{textAlign: 'center'}}>
            - TeamApp is a great tool for all of that!
          </p>
          <img src={require('./img/iphone.png')} alt='iphone demo'/>
        </section>
        <section>
          <h3>How to start?</h3>
          <p>
            Simple!
          </p>
          <ol>
            <li>Sign up for an account</li>
            <li>Set up a profile</li>
            <li>Create / Join a project</li>
            <li>Collaborate with a team</li>
          </ol>
          <img src={require('./img/join_demo.png')} alt='join a project demo'/>
        </section>
        <section>
            <Btn 
              onClick={handleGuestLogin}
              className={styles.guestBtn}
              type='button'
              title='Just curious? - Continue as a guest'/>
          <p className='devMsg'>
            Please note TeamApp is still in development. <br/>
            More features coming soon! <br/>
            If you'd like to contribute to the project, please visit our
            <a href='https://github.com'> github page</a>
          </p>
        </section>
        <footer>
          <a href='https://github.com'>{'\u00a9'} 2019 Alec Kan</a>
        </footer>
      </div>
    </div>
  </>
}

export default LandingPage