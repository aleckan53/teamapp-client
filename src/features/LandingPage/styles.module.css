.version {
  color: var(--color2);
  position: absolute;
  top: 4%;
  left: 5%;
  font-family: Verdana, Geneva, Tahoma, sans-serif;
  font-size: .85rem;
}

.landingPage > header {
  position: relative;
  box-shadow: 0px 5px 40px -2px var(--box-shadow);
}

.landingPage section {
  font-size: 1.4rem;
  line-height: 1.5
}

@media (min-width: 500px) {
  .landingPage > header {
    margin-top: -1%;
  }
  .version {
    margin-top: 1rem;
    left: 10%;
  }
}

.hero {
  background-image: linear-gradient(rgba(0,0,0,0) 65%, rgba(0,0,0,.8) 100%), url('./img/main.png');
  background-size: 150%;
  background-position: center;
  background-repeat: no-repeat;
  width: 100%;
  height: 400px;
}

@media (min-width: 1100px) {
  .hero {
    background-size: 100%;
  }
}

.btnContainer {
  position: absolute;
  display: flex;
  width: 40%;
  justify-content: space-between;
  top: 2%;
  right: 5%;
}

@media (min-width: 500px) {
  .btnContainer {
    margin-top: 1rem;
    right: 10%;
  }
}

.btn {
  font-size: .7rem;
  width: 45%;
  height: 2rem;
  text-align: center;
  padding: 9px 0;
}

@media (min-width: 700px) {
  .btnContainer {
    width: 30%;
  }
  .btn {
    margin-top: auto;
    margin-left: auto;
    right: 10%;
  }
}

.btn:hover {
  transform: scale(1.1,1.1)
}

.titleContainer {
  position: absolute;
  bottom: 0;
  left: 5%;
}

@media (min-width: 500px) {
  .titleContainer {
    left: 10%;
  }
}

@media (min-width: 1100px) {
  .btnContainer {
    right: 25%;
    width: 25%;
  }
  .titleContainer {
    left: 25%;
  }
  .version {
    left: 25%;
  }
}

@media (min-width: 1500px) {
  .btnContainer {
    right: 32%;
    width: 20%;
  }
  .titleContainer {
    left: 32%;
  }
  .version {
    left: 32%;
  }
}

.titleContainer h1 {
  color: var(--sub-text);
  margin-bottom: .4rem;
}

.titleContainer h2 {
  color: var(--main-color);
  margin-top: 0;
  font-size: 1.3rem;
}

.landingPage img {
  width: 100%;
  height: 100%;
}

.landingPage h3 {
  color: var(--main-text);
}

.guestMsg {
  color: var(--main-color);
  font-size: 1rem;
  text-align: center;
}

.guestBtn {
  margin-top: .5rem;
}

.landingPage ul, ol{
  padding-left: 25px;
  color: var(--main-text);
  list-style: none;
  font-weight: 600
}

.landingPage ul li::before {
  content: "\2022";
  color: var(--main-color);
  font-weight: bold;
  display: inline-block;
  width: 1em;
  margin-left: -25px;
}

.landingPage ol li {
  counter-increment: default;
}
.landingPage ol li::before {
  content: counter(default) ". ";
  color: var(--main-color);
  font-weight: bold;
  margin-left: -25px;
  margin-right: 10px;
}

.landingPage a {
  color: var(--main-text)
}

.landingPage li {
  margin: 5px 0;
}

footer {
  margin-top: 2rem;
  height: 60px;
  border-top: .6px solid var(--box-shadow);
  text-align: center;
  padding: 40px 0 0;
  font-size: .7rem
}

.landingPage button {
  margin-bottom: 3rem;
}

.landingPage button:hover {
  box-shadow:0px 0px 10px var(--main-color);
}

