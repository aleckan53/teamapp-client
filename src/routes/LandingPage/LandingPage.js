import React, { useContext, useState } from 'react'
import styles from './LandingPage.module.css'
import { BtnLink, Btn, LoaderFull } from '../../components/Basic/Basic'
import ipad from './files/ipad_2.png'
import phones from './files/3-phones.png'
import joinDemo from './files/join_demo.png'
import AuthApiService from '../../services/AuthApiService'
import EventsContext from '../../context/EventsContext'

const LandingPage = props => {
  const [loading, setLoading] = useState(false)
  const { setAuthorized } = useContext(EventsContext)

  const handleGuestLogin = () => {
    AuthApiService.guestLogin(setAuthorized, props.history, setLoading)
  }

  return (
    <>
    {loading ? <LoaderFull/> : ''}
    <div className={styles.landingPage}>
      <header className={styleMedia.header}>
        <div className={styles.hero}>
        </div>
        <span className={styles.version}>v 0.1 <strong>Beta</strong></span>
        <div className={styles.btnContainer}>
          <BtnLink 
            style={{color: 'var(--sub-text)'}}
            to='signup'
            className={styles.btn}
            type='button'
            title='Sign up'/>
          <BtnLink
            to='/login' 
            className={styles.btn}
            type='button'
            title='Log in'/>
        </div>
        <div className={styles.titleContainer}>
          <h1>A new way to collaborate!</h1>
          <h2>Welcome to the TeamApp</h2>
        </div>
      </header>
      <div>
        <section>
          <h3>About</h3>
          <p>
            TeamApp is a new collaboration platform, helping people to team up and work together.<br/><br/>
          </p>
          <img src={ipad} alt=''/>
          <p>
            If you want to:
          </p>
            <ul>
              <li>Bring your great idea into life</li>
              <li>Volunteer to an open source project</li>
              <li>Gain a real working experience</li>
              <li>Level up your skills</li>
            </ul>
          <p style={{textAlign: 'center'}}>
            - TeamApp is a great tool for all of that!
          </p>
          <img src={phones} alt=''/>
        </section>
        <section>
          <h3>How to start?</h3>
          <p>
            Simple!
          </p>
          <ol>
            <li>Sign up for an account</li>
            <li>Set up a profile</li>
            <li>Create / Join a project</li>
            <li>Collaborate with a team</li>
          </ol>
          <img src={joinDemo} alt=''/>
        </section>
        <section>
            <Btn 
              onClick={() => handleGuestLogin()}
              className={styles.guestBtn}
              type='button'
              title='Just curious? - Continue as a guest'/>
          <p className='devMsg'>
            Please note TeamApp is still in development. <br/>
            More features coming soon! <br/>
            If you'd like to contribute to the project, please visit our
            <a href='https://github.com'> github page</a>
          </p>
        </section>
        <footer>
          <a href='https://github.com'>{'\u00a9'} 2019 Alec Kan</a>
        </footer>
      </div>
    </div>
    </>
  )
}

export default LandingPage