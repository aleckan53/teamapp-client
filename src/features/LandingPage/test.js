import React from 'react'
import ReactDOM from 'react-dom'
import LandingPage from './index'
import { create } from 'react-test-renderer'
import { Route, BrowserRouter } from 'react-router-dom'

describe('LandingPage', () => {
  it('renders without crushing', () => {
    const div = document.createElement('div')
    ReactDOM.render(
      <BrowserRouter>
        <Route><LandingPage/></Route>
      </BrowserRouter>      
      , div)
    ReactDOM.unmountComponentAtNode(div)  
  })

  it('matches the snapshot', () => {
    const tree = create(
      <BrowserRouter>
        <Route><LandingPage/></Route>
      </BrowserRouter>
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

})