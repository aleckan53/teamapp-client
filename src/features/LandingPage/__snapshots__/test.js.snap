// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage matches the snapshot 1`] = `
Array [
  "",
  <div
    className="landingPage"
  >
    <header
      className="header"
    >
      <div
        className="hero"
      />
      <span
        className="version"
      >
        v 0.1 
        <strong>
          Beta
        </strong>
      </span>
      <div
        className="btnContainer"
      >
        <a
          className="btn btn"
          href="/signup"
          onClick={[Function]}
          style={
            Object {
              "color": "var(--sub-text)",
            }
          }
        >
          Sign up
        </a>
        <a
          className="btn btn"
          href="/login"
          onClick={[Function]}
        >
          Log in
        </a>
      </div>
      <div
        className="titleContainer"
      >
        <h1>
          A new way to collaborate!
        </h1>
        <h2>
          Welcome to the TeamApp
        </h2>
      </div>
    </header>
    <div>
      <section>
        <h3>
          About
        </h3>
        <p>
          TeamApp is a new collaboration platform, helping people to team up and work together.
          <br />
          <br />
        </p>
        <img
          alt="ipad demo"
          src="ipad.png"
        />
        <p>
          If you want to:
        </p>
        <ul>
          <li>
            Bring your great idea into life
          </li>
          <li>
            Volunteer to an open source project
          </li>
          <li>
            Gain a real working experience
          </li>
          <li>
            Level up your skills
          </li>
        </ul>
        <p
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          - TeamApp is a great tool for all of that!
        </p>
        <img
          alt="iphone demo"
          src="iphone.png"
        />
      </section>
      <section>
        <h3>
          How to start?
        </h3>
        <p>
          Simple!
        </p>
        <ul>
          <li>
            Sign up for an account
          </li>
          <li>
            Set up a profile
          </li>
          <li>
            Create / Join a project
          </li>
          <li>
            Collaborate with a team
          </li>
        </ul>
        <img
          alt="join a project demo"
          src="join_demo.png"
        />
      </section>
      <section>
        <button
          className="btn guestBtn "
          onClick={[Function]}
        >
          Just curious? - Continue as a guest
        </button>
        <p
          className="devMsg"
        >
          Please note TeamApp is still in development. 
          <br />
          More features coming soon! 
          <br />
          If you'd like to contribute to the project, please visit our
          <a
            href="https://github.com"
          >
             github page
          </a>
        </p>
      </section>
      <footer>
        <a
          href="https://github.com"
        >
          Â©
           2019 Alec Kan
        </a>
      </footer>
    </div>
  </div>,
]
`;
